apps/nextjs/src/components/activity/activityform.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from "react";
+import DatePicker from "react-datepicker";
+import { Controller, useForm } from "react-hook-form";
+
+import { api } from "~/utils/api";
+
+interface ActivityData {
+  name: string;
+  duration: string;
+  leader: string;
+  location: string;
+  selectedDate: Date;
+  startDate: Date;
+}
```

Comment:
you can use https://github.com/uoftblueprint/centre-for-dreams/blob/main/apps/nextjs/src/utils/api.ts#L69 for the type that the endpoint accepts

Reply:
i dont know how to do that i asked chatgpt and im still confused, do i define my own type inside the api file and then infer it in the activity form file? 😭

Reply:
<img width="464" alt="image" src="https://github.com/uoftblueprint/centre-for-dreams/assets/20374181/e4d1aa5b-1fa1-49ef-b9f9-7e0833e436ed">


like this -- this is similar to what you can do for the other component PRs (except those are using the outputs)